src/udev/udevadm-info.c

```diff
@@ -982,55 +1135,64 @@ int info_main(int argc, char *argv[], void *userdata) {
                         if (!path)
                                 return log_oom();
 
-                        r = strv_consume(&devices, path);
+                        r = strv_consume(&arg_devices, path);
                         if (r < 0)
                                 return log_oom();
                         break;
                 }
 
                 case 'q':
-                        action = ACTION_QUERY;
+                        arg_action = ACTION_QUERY;
                         if (streq(optarg, "property") || streq(optarg, "env"))
-                                query = QUERY_PROPERTY;
+                                arg_query = QUERY_PROPERTY;
                         else if (streq(optarg, "name"))
-                                query = QUERY_NAME;
+                                arg_query = QUERY_NAME;
                         else if (streq(optarg, "symlink"))
-                                query = QUERY_SYMLINK;
+                                arg_query = QUERY_SYMLINK;
                         else if (streq(optarg, "path"))
-                                query = QUERY_PATH;
+                                arg_query = QUERY_PATH;
                         else if (streq(optarg, "all"))
-                                query = QUERY_ALL;
+                                arg_query = QUERY_ALL;
                         else
                                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "unknown query type");
                         break;
+
                 case 'r':
                         arg_root = true;
                         break;
+
                 case 'd':
-                        action = ACTION_DEVICE_ID_FILE;
-                        r = free_and_strdup(&name, optarg);
+                        arg_action = ACTION_DEVICE_ID_FILE;
+                        r = free_and_strdup(&arg_name, optarg);
                         if (r < 0)
                                 return log_oom();
                         break;
+
                 case 'a':
-                        action = ACTION_ATTRIBUTE_WALK;
+                        arg_action = ACTION_ATTRIBUTE_WALK;
                         break;
+
                 case 't':
-                        action = ACTION_TREE;
+                        arg_action = ACTION_TREE;
                         break;
+
                 case 'e':
-                        action = ACTION_EXPORT;
+                        arg_action = ACTION_EXPORT;
                         break;
+
                 case 'c':
-                        cleanup_db();
-                        return 0;
+                        arg_cleanup_db = true;
```

Comment:
I think it is better to make this an action.

Reply:
Makes sense.